File: kala-game-service/src/main/java/org/jesperancinha/games/kalagameservice/dto/converters/PlayerConverter.java
package org.jesperancinha.games.kalagameservice.dto.converters;

import org.jesperancinha.games.kalagameservice.dto.PlayerDto;
import org.jesperancinha.games.kalagameservice.model.Player;

import java.util.Objects;

public class PlayerConverter {
    public static PlayerDto toDto(Player currentPlayer) {
        if (Objects.isNull(currentPlayer)) {
            return null;
        }
        return PlayerDto.builder()
                .id(currentPlayer.getId())
                .username(currentPlayer.getUsername())
                .build();
    }
}

File: kala-game-service/src/main/kotlin/org/jesperancinha/games/kalagameservice/dto/converters/PlayerConverter.kt
package org.jesperancinha.games.kalagameservice.dto.converters

import org.jesperancinha.games.kalagameservice.dto.PlayerDto
import org.jesperancinha.games.kalagameservice.model.Player

object PlayerConverter {
    fun toDto(currentPlayer: Player?): PlayerDto? {
        return currentPlayer?.let {
            PlayerDto(
                id = currentPlayer.id,
                username = currentPlayer.username
            )
        }
    }
}